go-controller/pkg/ovn/policy.go

```diff
@@ -195,7 +195,7 @@ func (oc *Controller) updateStaleDefaultDenyACLNames(npType knet.PolicyType, gre
 	p := func(item *nbdb.ACL) bool {
 		return item.ExternalIDs[defaultDenyPolicyTypeACLExtIdKey] == string(npType) && // default-deny-policy-type:Egress or default-deny-policy-type:Ingress
 			strings.Contains(item.Match, gressSuffix) && // Match:inport ==	@ablah80448_egressDefaultDeny or Match:inport == @ablah80448_ingressDefaultDeny
-			!strings.Contains(*item.Name, arpAllowPolicySuffix) && // != name: namespace_ARPallowPolicy
+			!strings.Contains(item.Match, "arp") && // != match: (arp || nd)
```

Comment:
To discard already migrated ACLs, why not filter out names that end with a substring of `gressSuffix` after the `_`. Otherwise we would always be migrating the deny ACLs with long namespace names.

Reply:
yeah I know we will be re-doing some of the work always on restart, but I felt that was a small price to pay since I thought it would be a corner case. TBH I am very hesitant to add back any filtering based on names or suffixes that go into the names. Example for the corner case 63 we will always redo migration for those acls.

Let me think a bit more if this is worth the complexity or not. But I have been thinking of this since yesterday... Maybe the match after `_` is ok since I checked that in K8s one cannot have a `_` in a namespace name. And along with the other three matches... maybe I am ok to consider the names substring match after the `_`. I wonder if just checking for `_` is good enough? Because since we rule out all arp allow acls and the first two match conditions will ensure we don't pick non-default acls.. that might work. @jcaamano : shall we just do:
1) a `_` match?
2) or does it really add value when we match on the `_e`, `_eg`, `_egre` and so on?
3) or we decide this is corner case maybe 10 namespaces in the cluster will always be migrated on restart?

Reply:
update: let me push an update for this and we can talk post that.

Reply:
The last `_` should be the one that we added provided there is no other `_` in the suffix. Also any other suffix we use for these ACLs should not share any common sub-string with this suffix. Both conditions hold true right now and we can probably document where we are defining them the matter that we rely on these facts.

If the namespace name ends with our suffix sub-string and is at the maximum length then it would be a noop in any case.

We can also test if the new name is equal to the old name and omit the update in that case.

I think that the most easy thing to do is to check that our suffix is a prefix of the ACL name suffix.


If we suspect a corner case we can test for it.

Reply:
sounds good, I have fixed this. Let me add some tests for the prefix in suffix logic that is standalone.

Reply:
done!